Rewrite HTTP OAuth2 client BBEs (#3793)
ballerina-platform/ballerina-distribution: examples/http-client-oauth2-client-credentials-grant-type/http_client_oauth2_client_credentials_grant_type.md

# HTTP client - OAuth2 client credentials grant type

> The `http:Client` can be secured with OAuth2 client credentials grant type authorization by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for this grant type can be specified in the `auth` field of the client configuration.

Author: ```suggestion
The `http:Client` can connect to a service that is secured with the OAuth2 client credentials grant type by adding the `Authorization: Bearer <token>` header to each request. The required configurations for this grant type can be specified in the `auth` field of the client configuration.
```

> The `http:Client` can be secured with OAuth2 client credentials grant type authorization by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for this grant type can be specified in the `auth` field of the client configuration. Use this grant type when the client needs to access resources or services on behalf of itself rather than the user.

Author: ```suggestion
The `http:Client` can be secured with OAuth2 client credentials grant type authorization by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for this grant type can be specified in the `auth` field of the client configuration.
```

> 

Reviewer: Lets remove this part `Use this grant type when the client needs to access resources or services on behalf of itself rather than the user.`

Author: Fixed via https://github.com/ballerina-platform/ballerina-distribution/pull/3793/commits/373fd382554d45179c4ec44e8ea2280b4b8709b2

> An `http:Client` can be secured with OAuth2 client credentials grant type authorization. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 client credentials grant type can be specified in the `auth` field as the `http:OAuth2ClientCredentialsGrantConfig` record. Use this to communicate with the service, which is secured with OAuth2 authorization.

Reviewer 2: We could add the following to answer **When to use ?** question
Use when client application needs to access resources or services on behalf of itself rather than on behalf of a user.

Author: Added via [187d99d](https://github.com/ballerina-platform/ballerina-distribution/pull/3793/commits/187d99dffcd52f92e019e05f224ac4dc5e4a2e66)

> An `http:Client` can be secured with OAuth2 client credentials grant type authorization. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 client credentials grant type can be specified in the `auth` field as the `http:OAuth2ClientCredentialsGrantConfig` record. Use this to communicate with the service, which is secured with OAuth2 authorization.

Reviewer 3: ```suggestion
The `http:Client` can be secured with OAuth2 client credentials grant type authorization. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 client credentials grant type can be specified in the `auth` field as the `http:OAuth2ClientCredentialsGrantConfig` record. Use this to communicate with the service, which is secured with OAuth2 authorization.
```

> The `http:Client` can be secured with OAuth2 client credentials grant type authorization. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 client credentials grant type can be specified in the `auth` field as the `http:OAuth2ClientCredentialsGrantConfig` record. Use this to communicate with the service, which is secured with OAuth2 authorization.

Author: ```suggestion
The `http:Client` can be secured with OAuth2 client credentials grant type authorization. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 client credentials grant type can be specified in the `auth` field as the `http:OAuth2ClientCredentialsGrantConfig` record. Use this grant type when the client needs to access resources or services on behalf of itself rather than the user.
```

> An `http:Client` can be secured with OAuth2 client credentials grant type authentication. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 client credentials grant type can be specified in the `auth` field as the `http:OAuth2ClientCredentialsGrantConfig` record. Use this to communicate with the service, which is secured with OAuth2 authentication.

Reviewer 2: oauth2 is used for authorization (not for authentication)

Author: Fix via [30a1a64](https://github.com/ballerina-platform/ballerina-distribution/pull/3793/commits/30a1a645001e203cffbbca292c0b0a5d7d34f335)

> The `http:Client` can be secured with OAuth2 client credentials grant type authorization. This is achieved by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for the OAuth2 client credentials grant type can be specified in the `auth` field as the `http:OAuth2ClientCredentialsGrantConfig` record. Use this grant type when the client needs to access resources or services on behalf of itself rather than the user.

Author: ```suggestion
The `http:Client` can be secured with OAuth2 client credentials grant type authorization by enriching each request from the client with the `Authorization: Bearer <token>` header. The required configurations for this grant type can be specified in the `auth` field of the client configuration. Use this grant type when the client needs to access resources or services on behalf of itself rather than the user.
```
